### PR TITLE
Add tests against multiple Nodejs versions

### LINKED ISSUE #7169: Add tests against multiple Node.js versions
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
Parse Server CI currently only tests against Node 10.0.0 and 12.y.z (latest patch version).
This does not reflect the variety of Node versions in use in real word deployments.

### Feature / Enhancement Description
Keep the Node versions for CI tests in sync with the Node versions in [Long-Term-Support](https://github.com/nodejs/node/blob/master/CHANGELOG.md).

That means:
- Removing the CI tests for Node 10 in April (EOL date April 2021).
- Add tests for the following LTS versions:

  | Version     | EOL Date     | Latest Patch Release |
  |-------------|--------------| - |
  | ~~Node 10~~ | ~~April 2021~~   |~~10.23.2~~ |
  | Node 12 | April 2022 | 12.20.1 |
  | Node 14 | April 2023          | 14.15.4 |

- When a new patch version is released, update the CI test version as well.

In addition, the Readme should contain a Node compatibility table to show against which Node versions Parse Server is tested.

This is related to https://github.com/parse-community/parse-server/issues/7160 which adds tests against multiple MongoDB versions and automatically ensures that tests always run against the latest patch version. A similar mechanism should be implemented for Node and leverage the existing mechanism.

### Example Use Case
(none)

### Alternatives / Workarounds
(none)

### 3rd Party References
(none)

### PR DESCRIPTION
**Pull request by mtrezza**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Parse Server is not tested against latest Nodejs versions.

Related issue: closes #7169

### Approach
Adding CI test script that automatically checks whether
- test environments run the latest Node.js patch version
- there is a newer Node.js minor / major version against which Parse Server is not tested

### TODOs before merging

- [x] Add entry to changelog
- [x] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by dplewis** [comment]
What about Node 15?

**Comment by mtrezza** [comment]
I may have misunderstood the [status](https://github.com/nodejs/Release) of 15 in the release plan, but I looked again and it seems you are right, we can add 15. It is actively being worked on while in release.

**Comment by mtrezza** [comment]
@dplewis I did more research and it confirms what you suggested:

According to [Node.js](https://nodejs.org/en/about/releases/):

> Major Node.js versions enter Current status for six months, which gives library authors time to add support for them. (...) Production applications should only use Active LTS or Maintenance LTS releases.

- Node 15 is now in Current status, the period in which we should get Parse Server ready for it
- Node 16 is expected to become Current in April.

That also implies that the default Node.js version for tests may well be bumped to Node 14, which is in Active long term support and the most recent version that is recommended for production apps.

image

File: README.md [hunk 1]
```diff
@@ -103,15 +103,25 @@ Before you start make sure you have installed:
...
-#### MongoDB Support
-Parse Server is continuously tested with the most recent releases of MongoDB to ensure compatibility. The rests run against the latest patch version of each MongoDB release. We follow the [MongoDB support schedule](https://www.mongodb.com/support-policy) and only test against versions that are officially supported by MongoDB and have not reached their end-of-life date yet.
+#### Node.js
+Parse Server is continuously tested with the most recent releases of Node.js to ensure compatibility. We follow the [Node.js Long Term Support plan](https://github.com/nodejs/Release) and only test against versions that are officially supported and have not reached their end-of-life date yet.
+
+| Version    | Latest Patch Version | End-of-Life Date | Compatibility      |
+|------------|----------------------|------------------|--------------------|
+| Node.js 10 | 10.23.2              | April 2021       | ✅ Fully compatible |
+| Node.js 12 | 12.20.1              | April 2022       | ✅ Fully compatible |
+| Node.js 14 | 14.15.4              | April 2023       | ✅ Fully compatible |
```
Comment (dplewis) [inline README.md:113]: Should we update the table?

Comment (mtrezza) [reply README.md:113]: Ah, yes, added Node 15. Funny enough, end of life is in June already.

**Review by dplewis — APPROVED** [review]

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/5.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.0.0](https://github.com/parse-community/parse-server/releases/tag/5.0.0)

**Pull request merged**